src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -553,24 +561,25 @@ namespace OpenRCT2::PathFinding
     static uint8_t PeepPathfindGetMaxNumberJunctions(Peep& peep)
     {
         if (peep.Is<Staff>())
-            return 8;
+            return kMaxJunctionsStaff;
 
         auto* guest = peep.As<Guest>();
         if (guest == nullptr)
-            return 8;
+            return kMaxJunctionsStaff;
 
-        if (guest->PeepFlags & PEEP_FLAGS_LEAVING_PARK && guest->GuestIsLostCountdown < 90)
+        bool isLeavingPark = (guest->PeepFlags & PEEP_FLAGS_LEAVING_PARK) != 0;
+        if (isLeavingPark && guest->GuestIsLostCountdown < 90)
         {
-            return 8;
+            return kMaxJunctionsGuestLeavingParkLost;
         }
 
-        if (guest->HasItem(ShopItem::Map))
-            return 7;
+        if (isLeavingPark)
```

Comment:
This changes the order in which these checks are done, is that intentional?

Reply:
I think this is more logical, the outcome is currently the same as the limit is identical for guest with map and guest who is leaving the park (not lost). At the end I don't think doing this is a good idea in general but if I change the logic it will influence the tests.

Reply:
I’m a bit apprehensive that the tests might miss something, considering @AaronVanGeffen just solved a bug that was very subtle and hard to diagnose, but had a devastating effect, hence my question.

Reply:
The function will result the same as before even with the changed logic which in my opinion is better structured.

Reply:
Well, I sincerely hope you’re right, then. The new logic certainly makes sense.

Reply:
This part of the code is one of the very first things it uses so if there is an effect the bpb replay test would be screaming now, I made sure that each commit pushed passed the tests.